### ISSUE TITLE
Feature Request: (2) new addons to also serve as developer examples/templates

### ISSUE THREAD

**Issue by CollinChaffin** [tablacus/TablacusExplorerAddons#43]
Hi Gaku,

Per my other issue I indicated I would open this issue separately.  It consists of two request items that go hand-in-hand:

----

## 1. New Column Population: SPACE FREE

What I am requesting you consider is an addon that is almost identical to the new Junction addon that looks ahead to locate the target of that selected item to then populate the item column.  It would do the following:

- Utilizing the new/improved free space API method you just added to the free space addon, populate the currently unused column named `Space Free` strictly for container (drive or directory) items and leave it null for file items.  In other words, pre-fetch what the "Size Status Bar" would produce if the user navigated into the selected container item, and use that prefetched number for the new `Space Free` on it saving the user from having to go into that target to view it's free space.

This would be a perfect companion to the Junction target for folks that have lots of either local or remote mapped drives, junctions, DFS, etc. to be able to quickly look at a list of remote junctions and add that new unused column to their view and instantly see each target's free space WITHOUT having to navigate into every single target one at a time and looking at a status bar but instead for the FIRST TIME in any explorer replacement, see it in an actual proper column.

----

## 2. Generate/Save Target Container Directory Listing to TXT/CSV

As a secondary addon that compliments the "Space Free", a pretty simple addon I can think of that is surprisingly still missing from TE that would also require this exact same "lookahead" code is one that:

- Generates and saves a text/CSV/Tab separated file containing a simple directory list when the user right-clicks on a container object of Drive/Directory after prompting with a simple file save dialog. 

----

Boy, if both of these were added but coded to be true "examples" taking time for lots of extra code comments and structuring it in a much more verbose manner I know it will almost definitely ensure I can begin contributing compatible addons and PR's, and I would bet I would not be the only one!  Given how similar they are to the Junction target link lookahead and literally use pretty much all existing major methods I'm guessing this would be very easy and quick for you and if you do please consider doing a bit of hand-holding and adding the additional code comments etc. as you know as a DEV it really does go a long way in understanding other DEV's code. :)

Thanks!

-C

**Comment by tablacus** [comment]
I will try to do future examination on that.
However, the "Free space" column exists only in "This PC".


Regards,

**Comment by CollinChaffin** [comment]
Yes sorry I meant "Space Free" which is BLANK on all my machines in TE so how did you get it to populate?

**Comment by CollinChaffin** [comment]
Excellent work Gaku!  I am testing right now but I think it looks perfect as usual!  I can start helping with some of the addons that are self explanatory but in other open issue I pointed out there are still quite a few that have no info and one or two base template addons with comments will really help other DEVs begin creating them with you!

**Comment by CollinChaffin** [comment]
That newest foldersettings addon needs to be made higher priority to first override the settings in remember.xml, but then IF that addon is also active, immediately alter the setting inside of remember.xml to whatever is defined in foldersettings.  Right now, it is the opposite, remember.xml is last and highest priority, so even if a folder view is reset it immediately writes to the remember.xml, and nothing past that can then change it.

Hope you understand the issue let me know if you do or if I need to explain more.

**Comment by ccfs** [comment]
"Remember folder view settings", "Take over folder view settings" and "Folder settings" add-ons are used to apply a folder view when you navigate, but with different mechanism. Therefore usually only one of them can be enabled anytime to avoid interference.

I suggest you to:
1. Disable all of them, then try visiting different folders (file folders / special folders / result lists) and changing the current view (view mode / visible columns / sorting / grouping).
2. Delete "remember.xml", enable only "Remember folder view settings", then re-try.
3. Delete "takeoverfoldersettings.txt", enable only "Take over folder view settings", then re-try.
4. Delete "foldersettings.xml", enable only "Folder settings", then re-try. You will need to choose "View > Customize Folder" to add/update some records.

Then you will have a better understanding of what these add-ons do, and will find that "Folder settings" gives you the most predictable results.

Before you understand their functions, I oppose to make any change to these add-ons.
